### PR TITLE
Updates SoyToIncrementalDomSrcCompiler

### LINKED ISSUE #4: Doesn’t work correctly with files with windows-style line-endings (CRLF)
We found the issue developing a Liferay tag(lib) with Soy. You can replicate the issue in a Liferay module and forcing the line-endings to be CRLF:

```
$ unix2dos src/main/resources/META-INF/resources/MyTemplate.soy
$ gradle clean build
... ... ...
> Building 67% > :modules:my-project:transpileJS
Running 'build'...
Compiling soy

stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^
Error: Compile error:
errors during Soy compilation
In file drop_area/DropArea.soy:1:23: parse error at '
': expected eof, {alias, {deltemplate, or {template
{namespace MyTemplate}
                      ^


:modules:tools-osgi:adr-frontend-taglib:transpileJS FAILED
... ... ...
BUILD FAILED

Total time: 17.777 secs
```

And obviously you can replicate the fix:

```
$ dos2unix src/main/resources/META-INF/resources/MyTemplate.soy
$ gradle clean build
... ... ...
BUILD SUCCESSFUL

Total time: 18.755 secs
```

### PR DESCRIPTION
**Pull request by jbalsas**
Hey guys!

This is an update of `SoyToIncrDomCompiler` to https://github.com/google/closure-templates/commit/7dae5a1a2b6868da939e7643741b67db1610f190 that fixes #4 along with many other things.

/cc @mairatma, @yuchi

### CODE REVIEW

**Comment by eduardolundgren** [comment]
Hey @jbalsas did you run all metal and metal-components tests using this version?

**Comment by jbalsas** [comment]
I'm on it right now, was trying to figure out https://github.com/metal/metal-tools-soy/issues/7 before...

**Comment by brunobasto** [comment]
Hey @jbalsas, I see they changed the JS dependency too: https://github.com/google/closure-templates/commit/7dae5a1a2b6868da939e7643741b67db1610f190#diff-68b90265f2e009fdb09b2c302a59838bL194

do we also need to update the JS deps we manually built from the previous version  here: https://github.com/eduardolundgren/metal-soy-bundle ?

**Comment by jbalsas** [comment]
You're probably right, @brunobasto, I'm trying to set it up to run the tests on `metal.js` right now...

**Comment by jbalsas** [comment]
Yeah, we definitely need to update it, although the valid one is actually https://github.com/metal/metal.js/tree/master/packages/metal-soy-bundle

I'll take a look, but the changes are quite big...

**Comment by brunobasto** [comment]
@jbalsas Yeah, I now realized I posted the wrong link, sorry :)

Since we are talking about it.. It would save us a lot of time on the future if we could automate the process of creating that JS file. I tried that while ago using the closure compiler but, not only it didn't work, but the generated file was also a bit larger than the one we have now. I think @mairatma  also tried that for some time.

**Comment by jbalsas** [comment]
Yeah, I was discussing it with @eduardolundgren right now... we should probably at least separate the things we're bringing in their different files, so diffing them in the future is easier, and the automate at least the concat and build part.

I'll ping you guys soon to discuss about it :)

**Comment by jbalsas** [comment]
I'm reopening since this is actually all we need to support the new compiler... I've sent https://github.com/metal/metal.js/pull/221 which should add support for it from metal-cli.

We can do a major version change here, to `4.0.0`, and we'll only start using it in `metal-cli` when the updated `soy-bundle` is merged. That way we should be safe from regressions since no one would be picking these tools to compile unless ready.

**Comment by eduardolundgren** [comment]
@jbalsas All tests are passing -- including all metal-components? If they are we are good to merge.

**Comment by jbalsas** [comment]
Let me run them all today, just to be sure :)

**Comment by jbalsas** [comment]
Hey @eduardolundgren, ~~I didn't have much time today, but I've been able to test the following (with the following steps, which are not easy due to how everything is bundled).~~~

Below you can see the necessary steps and result of the tests of `metal.js` and `metal-*` with this compiler version. 

TL;DR: Everything is green ☺️. Even if it weren't, we'll release a new major of `metal-tools-soy`, as well as a major of `gulp-metal` and `metal-cli`, so modules would need to opt-in into this making it less scary...

## metal.js ✅ 
- Delete `node_modules/metal-tools-soy`
- Run `npm link metal-tools-soy`
- [Update `gulpfile.js`](https://github.com/metal/metal.js/commit/786208ec20a1fdb52b03782d08708e09114d5914) for the newer version

## metal-components ☢️   

- Replace `node_modules/metal-incremental-dom` with the master version contents to fix https://github.com/metal/metal.js/issues/222. After this, tests are green. --> This is optional, depending if the tests are affected or not by it.
- Delete `node_modules/gulp-metal/node_modules/metal-tools-soy`
- Run `npm link metal-tools-soy` from `node_modules/gulp-metal`
- Replace `node_modules/metal-soy-bundle/build/bundle.s` with the master version file

component | tests
--- | --- 
metal-alert | ✅ 
metal-autocomplete | ✅ 
metal-autocomplete-badges | ✅ 
metal-button-group | ✅ 
metal-datatable | ✅  
metal-dropdown | ✅  
metal-layout | ✅  
metal-list | ✅  
metal-modal | ✅  
metal-pagination | ✅  
metal-popover | 🔴 (Tests already failing in metal-popover@master)
metal-progressbar | ✅  
metal-rating | ✅  
metal-reading-progress | ✅  
metal-router | ✅  
metal-select | ✅  
metal-slider | ✅  
metal-switcher | ✅  
metal-tabs | ✅  
metal-tooltip | ✅  
metal-treeview | ✅

**Comment by eduardolundgren** [comment]
@jbalsas Great news. Could you organize a minor release with those changes? Thanks!
